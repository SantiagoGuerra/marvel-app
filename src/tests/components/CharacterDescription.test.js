const React = require('react');
const { shallow } = require('enzyme');
const { default: CharacterDescription } = require('../../components/CharacterDescription');

describe('Test <CharacterDescription/> Component', () => {
  const thumbnail = {
    path: 'https://cloudimg.io/s/XSdTer23dwedffwqa',
    extension: 'png',
  };

  const name = 'Hulk';

  const description = 'lorem ipsum';

  const wrapper = shallow(
    <CharacterDescription
      thumbnail={thumbnail}
      name={name}
      description={description}
    />,
  );

  test('should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have [thumbnail, name and description] as parameters', () => {
    const wrapper = shallow(
      <CharacterDescription
        thumbnail={thumbnail}
        name={name}
        description={description}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
