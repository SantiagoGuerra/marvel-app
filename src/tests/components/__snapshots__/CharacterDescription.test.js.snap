// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CharacterDescription/> Component should have [thumbnail, name and description] as parameters 1`] = `
<Fragment>
  <Box
    height="medium"
  >
    <Image
      fit="cover"
      src="https://cloudimg.io/s/XSdTer23dwedffwqa.png"
    />
  </Box>
  <Box
    basis="80%"
    direction="column"
    gap="small"
  >
    <Heading
      level={1}
      responsive={true}
    >
      Hulk
    </Heading>
    <Text
      level={1}
    >
      lorem ipsum
    </Text>
  </Box>
</Fragment>
`;

exports[`Test <CharacterDescription/> Component should render the component correctly 1`] = `
<Fragment>
  <Box
    height="medium"
  >
    <Image
      fit="cover"
      src="https://cloudimg.io/s/XSdTer23dwedffwqa.png"
    />
  </Box>
  <Box
    basis="80%"
    direction="column"
    gap="small"
  >
    <Heading
      level={1}
      responsive={true}
    >
      Hulk
    </Heading>
    <Text
      level={1}
    >
      lorem ipsum
    </Text>
  </Box>
</Fragment>
`;
